Adds retry loops for transitional errors and specifies client hostname during init

ReubenMathew opened the pull request:
I just put 30 second timeouts for now.

ReubenMathew committed:
adds client hostname and username to client init

mprimi commented on internal/worker/worker.go:
```diff
@@ -111,13 +115,21 @@ func (w *workerImpl) processJob(job *core.JobRecord, revision uint64) (bool, err
-		return false, fmt.Errorf("Failed to update job %s: %v", job.Id, finalUpdateErr)
+	updateRetryTimer := time.NewTimer(updateRetryTimeout)
+finalUpdateLoop:
+	for {
```
Notice this is a 'busy loop': it retries as fast as possible. Which may spam the console with 1000s of errors per second

ReubenMathew replied:
Is a `1 * time.Second` retry delay good enough?

mprimi commented on internal/worker/worker.go:
```diff
@@ -4,14 +4,17 @@ import (
-	"github.com/mprimi/go-bench-away/internal/core"
-	"golang.org/x/sys/unix"
 	"io"
+	"log"
```
I don't think the rest of this project uses `log`

ReubenMathew replied:
Will change to `fmt`.

mprimi commented on internal/worker/worker.go:
```diff
@@ -85,10 +86,22 @@ func (w *workerImpl) processJob(job *core.JobRecord, revision uint64) (bool, err
-	if err != nil {
-		// TODO: retry if error is transitional
-		return false, fmt.Errorf("Failed to update job %s: %v", job.Id, err)
+	var newRevision uint64
```
Should this be squashed with the previous commit?

ReubenMathew replied:
@mprimi ready for review again!

mprimi commented on internal/worker/worker.go:
```diff
@@ -111,13 +128,21 @@ func (w *workerImpl) processJob(job *core.JobRecord, revision uint64) (bool, err
+		}
 	}
-
-	return false, nil
```
I think it's clearer if the final return is out here, rather than 2 levels nested

mprimi commented on internal/worker/worker.go:
```diff
@@ -4,22 +4,27 @@ import (
-	"github.com/mprimi/go-bench-away/internal/core"
-	"golang.org/x/sys/unix"
 	"io"
+	"log"
```
Still using log

mprimi commented on internal/client/dispatch.go:
```diff
@@ -88,10 +88,21 @@ dispatchLoop:
 
 		c.logDebug("Dispatching job %s", jobId)
 
-		// TODO implement retry
```
This TODO is probably deprecated. Handle is a local function so wrapping it into a retry like the rest doesn't make sense.

mprimi reviewed (commented):
Some changes may be missing. Still confused about the way commits are grouped and their message.

ReubenMathew commented:
> Some changes may be missing. Still confused about the way commits are grouped and their message.

@mprimi I tried to cleanup the commits. One for the worker retry logic, one for the hostname.

ReubenMathew committed:
adds retry logic for updating worker job status

adds a final job update retry loop for worker

adds a job revision update retry loop for worker
